go/cross-compilation.md

```diff
@@ -0,0 +1,92 @@
+Cross-compilation support in rules_go
+=====================================
+
+## Objective
+
+Provide support in rules_go for generating binary artifacts targeted
+to a different os/architecture than the host.
+
+## Use cases
+
+1. Generating go artifacts for distribution in external archives
+  targeted for a specific platform.
+
+2. Generating go artifacts for inclusion into docker images via the
+  `docker_build` rule.
+
+## Proposal
+
+The ideal scenario would:
+
+* Not require additional end-user rules, i.e. `go_binary` should work
+  for this purpose.
+
+* Allow the user to generate a cross-compiled go binary artifact
+  without modification of the `BUILD` file containing the `go_binary`
+  rule (use case #1).  This would likely work in concert with the
+  `--cpu` and/or `--crosstool_top` option.
+
+* Allow the user to explicity state the target architecture in the
+  `go_binary` rule, when the target architecture is a fixed/known
+  constant along a multistep build path (use case #2).
+
+## Implementation plan
+
+* Implement capture of absolute path of the go toolchain during
+  toolchain configuration (via the `go_root` rule).  For reasons
+  inherent to go itself, cross-compilation does not seem to work when
+  invoked via a non-absolute path (see
+  https://github.com/golang/go/issues/17017).
+
+* In `def.bzl`, `GOOS` and `GOARCH` are already included in the
+  environment of ctx.actions.  Add `GOROOT` into that dict, and change
+  all invocations of `go tool CMD` to use `$GOROOT/bin/go` rather than
+  `('../' * out_depth) + ctx.file.go_tool.path`.
+
+* Add a non-mandatory string attribute `os_arch` to the go_binary
```

Comment:
Would `os_arch` designate the target architecture?

Then this would mean that the Go rules would work differently from any other rule (which always target the target platform).